### PR TITLE
Search

### PR DESCRIPTION
**Pull request by nschloe**
Implements article search with typeahead in the navbar and user search with mentio in a couple of textarea fields.

### CODE REVIEW

File: src/templates/shared/navbar.html [hunk 1]
```diff
@@ -23,9 +23,26 @@
     <div class="collapse navbar-collapse" collapse="collapsed">
       <!-- left -->
       <div class="navbar-left">
-        <form class="navbar-form" role="search">
+        <form
+          name="searchForm"
+          class="navbar-form"
+          role="search"
+          ng-submit="searchForm.$valid && phSearch(search.body, 10)"
```
Comment (andrenarchy) [inline src/templates/shared/navbar.html]: do we need this?

File: src/templates/shared/navbar.html [hunk 2]
```diff
@@ -23,9 +23,26 @@
...
+              placeholder="Search PaperHub"
+              name="srch-term"
+              id="srch-term"
+              ng-model="search.body"
+              typeahead="result.title for result in phSearch($viewValue, 5)"
+              typeahead-wait-ms="250"
+              typeahead-on-select="goToArticle($item, $model, $label)"
+              required
+              >
             <div class="input-group-btn">
```
Comment (andrenarchy) [inline src/templates/shared/navbar.html]: maybe we should remove the button and place the icon in the input field (like [here](http://getbootstrap.com/css/#forms-control-validation))

File: src/js/controllers/navbar.js [hunk 3]
```diff
@@ -1,6 +1,29 @@
...
+            return response.data;
+          });
+        };
+
+        $scope.goToArticle = function(item, model, label) {
+          $location.path($routeSegment.getSegmentUrl(
+            'articles', {articleId: item._id}
+          ));
+          // for some reason, $route.reload() isn't enough here
+          $window.location.reload();
```
Comment (andrenarchy) [inline src/js/controllers/navbar.js]: it should even work without this line, see [here](https://github.com/paperhub/paperhub-frontend/blob/master/src/js/controllers/articleNew.js#L17)

Comment (nschloe) [inline src/js/controllers/navbar.js]: Well, it doesn't. :)

Comment (andrenarchy) [inline src/js/controllers/navbar.js]: Haha :) `location.reload()` is very un-dude because it reloads the whole document (like hitting the reload button).

Comment (nschloe) [inline src/js/controllers/navbar.js]: I realize that. However, I failed to find another solution to the problem.

Comment (andrenarchy) [inline src/js/controllers/navbar.js]: It works without the line from the landing page. I'm investigating... :)

**Comment by andrenarchy** [comment]
The `location.reload()` issue is fixed in the last commit.

I think we should move all search-related code from the `NavbarCtrl` into its own controller or a directive. What do you think?

**Comment by nschloe** [comment]
> I think we should move all search-related code from the NavbarCtrl into its own controller or a 
> directive. What do you think?

Fair enough.

**Pull request merged**